font/src/ft/mod.rs

```diff
@@ -68,7 +68,7 @@ pub struct FreeTypeRasterizer {
 
 #[inline]
 fn to_freetype_26_6(f: f32) -> isize {
-    ((1i32 << 6) as f32 * f) as isize
+    ((1i32 << 6) as f32 * f).ceil() as isize
```

Comment:
Why `ceil()` here? It seems like that would cause the same issue as floor, depending on input value.

Reply:
Well, the bug sounded like an off by one issue, so I dig around and came across this little conversion and thought that ceil semantics would actually be more correct.

This actually seemed to work so I googled further around to see if other projects were actually doing the same, and it turns out that at least cairo and kitty were also rounding up:

https://chromium.googlesource.com/chromiumos/third_party/cairo/+/master/src/cairo-ft-font.c#784 (cairo)
https://github.com/kovidgoyal/kitty/blob/master/kitty/freetype.c#L190 (kitty)

Besides that, I am not really an expert in this area, so I can't say for sure if this "should be the correct behavior".

Reply:
The cairo code you've linked isn't rounding up. It's rounding.

Adding 0.5 and then converting from float to int is a way to round for positive numbers.

floor(1.3 + 0.5) = 1
floor(ceil(1.3)) = 2

Reply:
Oh yes, smh, it's rounding. Though, it seems to "also" work with rounding (Looks like I was lucky.) 

Size: 9.5pt
px size: 9.5 * 96.0 / 72.0 ~= 12.666667

master: to_freetype_26_6(12.666667) = 810
rounding would produce 811, and that one difference really makes a diffenrence.

Well, yeah, I also not entirely sure if we are just shifting the issue around, but by actually rounding the value (to_freetype_26_6), we could at least get the same behavior as cairo and kitty (-ish since it's ceiling and not rounding).

Reply:
> we could at least get the same behavior as cairo and kitty (-ish since it's ceiling and not rounding).

As you've mentioned, Kitty is indeed ceiling and not rounding. So I'm a bit doubtful that this change would fix things in general, since both sizes should be valid.

Certain combinations of DPI and font size would still be able to produce any possible value here (since DPI can be any fraction). So I'd say this is almost guaranteed not to fix the issue in all cases.

The only change that could be reliable here would be to fix the font size to a multiple of 64 by rounding the font size before passing it to `to_freetype_26_6`. However I don't think anyone else is doing that, so I'm not sure if that's an issue.

Reply:
> The only change that could be reliable here would be to fix the font size to a multiple of 64 by rounding the font size before passing it to to_freetype_26_6. However I don't think anyone else is doing that, so I'm not sure if that's an issue.

I am not sure I quite follow. We wouldn't be able to specify fractional points (pixel?) then, since the fractional bits would always be zero, if the size would be a multiple of 64.

In my case, I actually just want to be able to use `12.5` as size forwarded to `set_char_size` and skip the "pt to px" calculation entirely because right now specifying `9.375` as size in alacritty would not be possible because `font::Size`'s granularity (factor 2) is too coarse.

Perhaps, another idea to "fix" this would be to completely skip the conversion and be able to specify the "px size". Or another hacky solution would be to increase the factor in `font::Size` to 8, so sizes like `9.375` can be represented which will turn to the desired `12.5` (9.375 * 96.0 / 72.0 = 12.5 -> * 64 = 800).

Reply:
Well, another quick & hacky would be to increase `font::Size`'s factor to 8 and change the font-size to `9.375`. That also works.